pkg/controller/controller.go

```diff
@@ -50,6 +50,7 @@ type controller struct {
 	startLock sync.Mutex
 
 	name        string
+	ctxID       string
```

Comment:
Should this be behind a lock? It's written once when creating the controller object, it's written a second time in `Start()` and we read it in another goroutine.

Reply:
It's already behind the `startLock` isn't it? And will only be assigned once (as `Start()` returns early if `c.started` is `true`. Also, the code that reads its value is not yet running, so we should be fine there too.

Reply:
You're right. In that case the write is done under a lock and the goroutine doing the read can never read the value while it's being written to.